### PR TITLE
Add docker-compose-plugin package for Debian

### LINKED ISSUE #61: Newer version of docker-compose required
# 🐛 Summary #

The latest version of the Reporting Engine AKA Pentest Portal requires `docker compose` version 2.2.3 or higher, but the corresponding COOL AMIs only have version 1.25.0 available.

See also [this Jira ticket](https://jira.ceil.cyber.dhs.gov/projects/CAL/queues/custom/20/CAL-18104).

## To reproduce ##

1. `docker-compose version`

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request adds the `docker-compose-plugin` package for Debian instances.

## 💭 Motivation and context ##

[The assessors require `docker compose` version 2.2.3 or later](https://jira.ceil.cyber.dhs.gov/projects/CAL/queues/custom/20/CAL-18104).  The `docker-compose` package currently supports a `1.x` version, while the `docker-compose-plugin` package supports a `2.x` version; therefore, the assessors' needs dictate that we install the `docker-compose-plugin` package.

The difference between the two Debian packages is that `docker-compose` provides a `docker-compose` command, while the `docker-compose-plugin` package provides `docker compose` functionality to the `docker` command.

## 🧪 Testing ##

All automated tests pass.  These changes were also tested successfully via the manual testing done in cisagov/docker-packer#41.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

**Comment by jsf9k** [comment]
I'm open to removing `docker-compose` anywhere we are able to install `docker-compose-plugin`.  Please let me know what you guys think is best.

**Comment by mcdonnnj** [comment]
> I'm open to removing `docker-compose` anywhere we are able to install `docker-compose-plugin`. Please let me know what you guys think is best.

I'm fine with that because Docker has made it pretty clear that [`docker compose` is the future](https://docs.docker.com/compose/#compose-v2-and-the-new-docker-compose-command). That being said we should check the [compatibility list](https://docs.docker.com/compose/cli-command-compatibility/) to make sure there's nothing breaking.

File: molecule/default/tests/test_default.py [hunk 1]
```diff
@@ -17,32 +17,67 @@ def test_packages(host):
...
-    elif distribution == "fedora":
-        # Only Moby is available for Feodra 32 and 33
-        #
-        # https://docs.docker.com/engine/install/fedora/
-        assert host.package("moby-engine").is_installed
+    if distribution in ["debian"]:
+        if codename in ["stretch"]:
+            assert all(
+                [
+                    host.package(pkg)
```
Comment (mcdonnnj) [inline molecule/default/tests/test_default.py]: I believe that as long as it is a valid package (and maybe even if it isn't) the `Package` class will be populated with other information and be truthy. I tested this by adding the following lines:

```python
assert host.package("irssi"), "no irssi package information found"
assert host.package("irssi").is_installed is False, "irssi is installed"
assert False, f"{host.package('irssi') is True} :: {host.package('irssi').is_installed} :: {all([host.package('irssi')])} :: {all([host.package('irssi').is_installed])}"
```

and the output when running `molecule test`:

```
>               assert False, f"{host.package('irssi') is True} :: {host.package('irssi').is_installed} :: {all([host.package('irssi')])} :: {all([host.package('irssi').is_installed])}"
E               AssertionError: False :: False :: True :: False
E               assert False
```

```suggestion
                    host.package(pkg).is_installed
```

File: molecule/default/tests/test_default.py [hunk 2]
```diff
@@ -17,32 +17,67 @@ def test_packages(host):
...
+                        "docker-ce",
+                        "docker-compose",
+                        "python-docker",
+                    ]
+                ]
+            )
+        elif codename in ["buster", "bullseye"]:
+            assert all(
+                [
+                    host.package(pkg)
```
Comment (mcdonnnj) [inline molecule/default/tests/test_default.py]:
```suggestion
                    host.package(pkg).is_installed
```

File: molecule/default/tests/test_default.py [hunk 3]
```diff
@@ -17,32 +17,67 @@ def test_packages(host):
...
+                        "docker-compose",
+                        "docker-compose-plugin",
+                        "python3-docker",
+                    ]
+                ]
+            )
+        elif codename in ["bookworm"]:
+            assert all(
+                [
+                    host.package(pkg)
```
Comment (mcdonnnj) [inline molecule/default/tests/test_default.py]:
```suggestion
                    host.package(pkg).is_installed
```

File: molecule/default/tests/test_default.py [hunk 4]
```diff
@@ -17,32 +17,67 @@ def test_packages(host):
...
+                    for pkg in ["docker.io", "docker-compose", "python3-docker"]
+                ]
+            )
+        else:
+            assert False, f"Unknown codename {codename}"
+    elif distribution in ["kali"]:
+        assert all(
+            [
+                host.package(pkg)
```
Comment (mcdonnnj) [inline molecule/default/tests/test_default.py]:
```suggestion
                host.package(pkg).is_installed
```

File: molecule/default/tests/test_default.py [hunk 5]
```diff
@@ -17,32 +17,67 @@ def test_packages(host):
...
+                for pkg in ["docker.io", "docker-compose", "python3-docker"]
+            ]
+        )
+    elif distribution in ["ubuntu"]:
+        assert all(
+            [
+                host.package(pkg)
```
Comment (mcdonnnj) [inline molecule/default/tests/test_default.py]:
```suggestion
                host.package(pkg).is_installed
```

File: molecule/default/tests/test_default.py [hunk 6]
```diff
@@ -17,32 +17,67 @@ def test_packages(host):
...
+                    "docker-compose",
+                    "docker-compose-plugin",
+                    "python3-docker",
+                ]
+            ]
+        )
+    elif distribution in ["amzn", "fedora"]:
+        assert all(
+            [
+                host.package(pkg)
```
Comment (mcdonnnj) [inline molecule/default/tests/test_default.py]:
```suggestion
                host.package(pkg).is_installed
```

**Review by mcdonnnj — CHANGES_REQUESTED** [review]

File: molecule/default/tests/test_default.py (hunk 1 above)
Comment (jsf9k) [reply molecule/default/tests/test_default.py]: Yes, I was rushing to wrap this up and copy-pasted the same foolishness all over the joint.  Thank you for cleaning it up.

**Comment by jsf9k** [comment]
> > I'm open to removing `docker-compose` anywhere we are able to install `docker-compose-plugin`. Please let me know what you guys think is best.
> 
> I'm fine with that because Docker has made it pretty clear that [`docker compose` is the future](https://docs.docker.com/compose/#compose-v2-and-the-new-docker-compose-command). That being said we should check the [compatibility list](https://docs.docker.com/compose/cli-command-compatibility/) to make sure there's nothing breaking.

I see nothing breaking.

File: molecule/default/tests/test_default.py [hunk 7]
```diff
@@ -52,6 +85,24 @@ def test_services(host, svc):
 
 
 @pytest.mark.parametrize("command", ["docker-compose version"])
```
Comment (mcdonnnj) [inline molecule/default/tests/test_default.py]:
```suggestion
```

File: molecule/default/tests/test_default.py [hunk 8]
```diff
@@ -52,6 +85,24 @@ def test_services(host, svc):
...
+            assert False, f"Unknown codename {codename}"
+    elif distribution in ["kali"]:
+        assert all([host.run(cmd).rc == 0 for cmd in ["docker-compose version"]])
+    elif distribution in ["ubuntu"]:
+        assert all([host.run(cmd).rc == 0 for cmd in ["docker compose version"]])
+    elif distribution in ["amzn", "fedora"]:
+        assert all([host.run(cmd).rc == 0 for cmd in ["docker-compose version"]])
+    else:
+        assert False, f"Unknown distribution {distribution}"
     assert host.run(command).rc == 0
```
Comment (mcdonnnj) [inline molecule/default/tests/test_default.py]:
```suggestion
```

**Review by mcdonnnj — COMMENTED** [review]
I believe these should be removed to resolve tests failing after your latest commit (and in concert with your refactor of how the testing is done).

File: molecule/default/tests/test_default.py (hunk 8 above)
Comment (jsf9k) [reply molecule/default/tests/test_default.py]: Already taken care of when I force-pushed the latest commit.

File: molecule/default/tests/test_default.py (hunk 7 above)
Comment (jsf9k) [reply molecule/default/tests/test_default.py]: Please see commit fb527a4.

Comment (mcdonnnj) [reply molecule/default/tests/test_default.py]: Ah, sorry for forgetting to adjust the function signature.

File: vars/Debian_stretch.yml [hunk 9]
```diff
@@ -1,6 +1,8 @@
 ---
 # The system packages to install
 package_names:
+  # Note that the docker-compose-plugin package is unavailable for
+  # Debian stretch.
```
Comment (mcdonnnj) [inline vars/Debian_stretch.yml]:
```suggestion
  # Debian Stretch.
```

**Review by mcdonnnj — APPROVED** [review]
Looking good! One small suggestion from me but otherwise :shipit:

File: molecule/default/tests/test_default.py [hunk 10]
```diff
@@ -51,7 +84,21 @@ def test_services(host, svc):
...
+        if codename in ["buster", "bullseye"]:
+            assert host.run("docker compose version").rc == 0
+        elif codename in ["stretch", "bookworm"]:
+            assert host.run("docker-compose version").rc == 0
+        else:
+            assert False, f"Unknown codename {codename}"
+    elif distribution in ["amzn", "fedora", "kali"]:
+        assert host.run("docker-compose version").rc == 0
+    elif distribution in ["ubuntu"]:
+        assert host.run("docker compose version").rc == 0
```
Comment (dav3r) [inline molecule/default/tests/test_default.py]: Perhaps a comment similar to this (from this PR's description) would be helpful to add in here:

> The difference between the two Debian packages is that docker-compose provides a docker-compose command, while the docker-compose-plugin package provides docker compose functionality to the docker command.

**Review by dav3r — APPROVED** [review]
👍  Though please consider my one comment request.

File: molecule/default/tests/test_default.py (hunk 10 above)
Comment (jsf9k) [reply molecule/default/tests/test_default.py]: Please see commit 0b18fb4.

Comment (dav3r) [reply molecule/default/tests/test_default.py]: Gracias!

**Pull request merged**